src/Core/PluginInterface.php

```diff
@@ -20,6 +20,9 @@
 
 /**
  * Plugin Interface
+ *
+ * @method \Cake\Core\ContainerInterface register(\Cake\Core\ContainerInterface $container) Register plugin services to
```

Comment:
Maybe this method should be declared as a part of separate `ContainerPluginInterface`?

Reply:
Would we then merge the two interfaces in 5.0? I thought an interface with a single method in it was a bit of a waste.

Reply:
I'd like to avoid interface method confusion we had in 3.x with entity methods being added to interface in docblocks.